### ISSUE TITLE
Consider releasing to NPM or adding precompiled dist or removing package.json "files" property

### ISSUE THREAD

**Issue by squareloop1** [fastify/fastify-passport#7]
First off: Great work! Been waiting for something like this and can't wait for the first release.

## 🚀 Feature Proposal

I am currently testing this package locally with NestJS and it is working great so far. However, it is kind of cumbersome to add this package as a dependency. It is currently not released to NPM and the "files" property in the package.json would require a precompiled "dist" directory to install this package directly via npm as a github dependency (it currently is just installing an empty directory with the readme and package.json as there is no "dist" directory).

We could handle this in one of three ways:

- Release to NPM
- Add precompiled dist to the repository
- Remove "files" property from package.json (which would require the consumer to compile it first)

## Motivation

To be able to test this dependency more comfortably by just installing it as a dependency.

**Comment by RobertLowe** [comment]
@squareloop1 I made a fork you should be able to install via giturl for now: https://github.com/fastify/fastify-passport/compare/master...RobertLowe:master

@fox1t any ETA on release / out of alpha?

@squareloop1 Im also working on using it in nestjs, im interested in how youve implemented it

**Comment by squareloop1** [comment]
@RobertLowe cool, I had a quick look at your implementation and I am basically doing exactly the same, except my `createPassportContext` method looks slightly different:
```
const createPassportContext = (request, response) => async (type, options, callback) => {
  return new Promise(async (resolve, reject) => {
    try {
      return await request.passport.authenticate(type, options, (request, response, err, user, info, status) => {
        try {
          request.authInfo = info;
          return resolve(callback(err, user, info, status));
        } catch (err) {
          reject(err);
        }
      })(request, response);
    } catch (error) {
      reject(error);
    }
  });
}
```

Currently struggling on how to add the fastify plugins (secureSession & `passport.initialize()`) in the dynamic module. Adding them in the `onModuleInit` didn't work.

I have forked the repo locally myself, but I'd rather have this package have some kind of release to comfortably add it as a dependency without the need to precompile first.

**Comment by RobertLowe** [comment]
@squareloop1 keep me posted, you can change:

```
const createPassportContext = (request, response) => async (type, options, callback) => {
  return new Promise(async (resolve, reject) => {
```
to 
```
const createPassportContext = (request, response) => {
  request.passport = passport;
  return async (type, options, callback) => {
```
to bypass the initialize check, there's definitely a better way to do it, i just don't know, I only picked up nestjs yesterday

No clue on secureSession's, keep me up to date 👍

**Comment by squareloop1** [comment]
I think we should move our discussion to keep this clear in here and prevent any confusion with the external development using this package that has nothing to do with this issue.

You can find me on the official fastify and nestjs discord. We could discuss the nestjs/fastify-passport development in more detail over there if you'd want to :)

**Comment by mcollina** [comment]
Released v0.1.0!
